read-fonts/src/tables/ift.rs

```diff
@@ -193,6 +228,92 @@ impl<'a> Iterator for GidToEntryIter<'a> {
     }
 }
 
+impl<'a> GlyphPatches<'a> {
+    /// Returns an iterator over the per glyph data for the table with the given index.
+    pub fn glyph_data_for_table(
+        &'a self,
+        table_index: u32,
+    ) -> impl Iterator<Item = Result<(GlyphId, &'a [u8]), ReadError>> {
+        let glyph_count = self.glyph_count() as usize;
+        let start_index = table_index as usize * glyph_count;
+        let start_it = self.glyph_data_offsets().iter().skip(start_index);
+        let end_it = self.glyph_data_offsets().iter().skip(start_index + 1);
```

Comment:
I wonder if `skip` is smart enough in this instance to avoid actually doing a linear scan, and just compiles down to `self.glyph_data_offsets()[start_index..].iter()`?